src/DynamoPublish/Models/InviteModel.cs

```diff
@@ -68,7 +69,20 @@ internal InviteModel(IAuthProvider dynamoAuthenticationProvider)
             if (String.IsNullOrWhiteSpace(invite))
                 throw new Exception(Resources.PageErrorMessage);
 
-            authenticationProvider = dynamoAuthenticationProvider;            
+            authenticationProvider = dynamoAuthenticationProvider;
+        }
+
+        internal string GetInvitationStatus() 
+        {
+            var req = new RestRequest(invite, Method.GET);
+
+            restClient = new RestClient(serverUrl);
+
+            authenticationProvider.SignRequest(ref req, restClient);
```

Comment:
If the user isn't already logged in, we should bail here and not update the status.